src/librustc_trans/debuginfo/gdb.rs

```diff
@@ -22,19 +23,13 @@ use std::ptr;
 use syntax::attr;
 
 
-/// Inserts a side-effect free instruction sequence that makes sure that the
-/// .debug_gdb_scripts global is referenced, so it isn't removed by the linker.
-pub fn insert_reference_to_gdb_debug_scripts_section_global(ccx: &CrateContext, builder: &Builder) {
+/// Inserts the .debug_gdb_scripts global into the set of symbols
+/// to be placed in `llvm.used`, so it isn't removed by the linker.
```

Comment:
So incidentally I was curious where did the DEBUGGING bit appear, since that's not something gcc (or gas) can do, and I looked at the binutils source code. It *looks* like all `.debug*` sections should have their SEC_ALLOC bit removed (and SEC_DEBUGGING bit added)...
```c
  if ((flags & SEC_ALLOC) == 0)
    {
      /* The debugging sections appear to be recognized only by name,
	 not any sort of flag.  Their SEC_ALLOC bits are cleared.  */
      if (name [0] == '.')
	{
	  const char *p;
	  int n;
	  if (name[1] == 'd')
	    p = ".debug", n = 6;
	  // [snip]
	  else
	    p = NULL, n = 0;
	  if (p != NULL && strncmp (name, p, n) == 0)
	    flags |= SEC_DEBUGGING;
	}
```
... but this is evidently not happening for `.debug_gdb_scripts` for some reason that I have no doubt is deeply unsettling in more than one way. There's really a large amount of logic in binutils that touches `SEC_DEBUGGING`, most of which is almost but not exactly identical platform code, and the more I look into it, the less I want to be aware of it. Or be conscious.